scout/server/extensions/clinvar_extension.py

```diff
@@ -77,10 +80,70 @@ def submit_json(self, json_data, api_key=None):
         except Exception as ex:
             return self.submit_service_url, None, ex
 
-    def show_submission_status(self, submission_id: str, api_key=None):
+    def json_submission_status(self, submission_id: str, api_key=None) -> dict:
         """Retrieve the status of a ClinVar submission using the https://submit.ncbi.nlm.nih.gov/api/v1/submissions/SUBnnnnnn/actions/ endpoint."""
 
         header: dict = self.set_header(api_key)
         actions_url = f"{self.submit_service_url}{submission_id}/actions/"
         actions_resp: requests.models.Response = requests.get(actions_url, headers=header)
-        flash(f"Response from ClinVar: {actions_resp.json()}", "primary")
+        return actions_resp.json()
+
+    def get_clinvar_scv_accession(self, url: str) -> Optional[str]:
+        """Downloads a submission summary from the given URL into a temporary file and parses this file to retrieve a SCV accession, if available."""
+        # Send a GET request to download the file
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an error if the request failed
+
+        # Create a temporary file to save the JSON content
+        with tempfile.NamedTemporaryFile(mode="w+", suffix=".json", delete=True) as temp_file:
+            # Write the JSON content to the temporary file
+            temp_file.write(response.text)
+            temp_file.flush()  # Ensure content is written to disk
+
+            # Read and parse the JSON content from the file
+            temp_file.seek(0)  # Reset file pointer to the beginning
+            json_data = json.load(temp_file)
+
+            submission_data: dict = json_data["submissions"][0]
+            processing_status: str = submission_data["processingStatus"]
+            if processing_status != "Success":
+                flash(
+                    f"Could not delete provided submission because its processing status is '{processing_status}'.",
+                    "warning",
+                )
+                return
+            return submission_data["identifiers"]["clinvarAccession"]
+
+    def delete_clinvar_submission(
+        self, submission_id: str, api_key=None
+    ) -> Optional[Tuple[int, dict]]:
+        """Remove a successfully processed submission from ClinVar."""
+
+        try:
```

Comment:
As safe as it can be, it it crashes at any point it just displays the error and redirects to the submissions page